src/olympia/shelves/forms.py

```diff
@@ -0,0 +1,51 @@
+import requests
+
+from rest_framework.reverse import reverse as drf_reverse
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+from olympia.shelves.models import Shelf
+
+
+class ShelfForm(forms.ModelForm):
+    class Meta:
+        model = Shelf
+        fields = '__all__'
+
+    def clean_criteria(self):
+        data = self.cleaned_data
+        criteria = data['criteria']
+        if (data['shelf_type'] == 'extension' or
+                data['shelf_type'] == 'theme' or
+                data['shelf_type'] == 'search'):
+            url = "https://addons.mozilla.org" + \
+                drf_reverse('v4:addon-search') + criteria
+        elif data['shelf_type'] == 'categories':
+            url = "https://addons.mozilla.org/api/v4/addons/categories/" + \
```

Comment:
do the same `drf_reverse` lookup` with these urls too.

Reply:
If you ever can't find the view name for a url then `./manage.py show_urls` lists them - the name is in red on the right hand side.